instance/models/instance.py

```diff
@@ -217,6 +220,21 @@ class GitHubInstanceQuerySet(models.QuerySet):
     Additional methods for instance querysets
     Also used as the standard manager for the GitHubInstance model
     """
+    def update_or_create_from_pr(self, pr):
+        """
+        Create or update an instance for the given pull request
+        """
+        pr_sub_domain = 'pr{number}.sandbox'.format(number=pr.number)
```

Comment:
Is there a way to avoid logic specific to sandboxes in instances models/managers? (ie the domain name, here) Sandboxes are a specific use case of instances, there might be PRs that aren't related to sandboxes in the future; like a PR against a release branch used for a production instance, for example.